scheduler/src/cook/rest/api.clj

```diff
@@ -912,11 +920,18 @@
                 (assoc :partitions (into #{} (walk/stringify-keys partitions))))))
        (into #{})))
 
+(defn validate-job-gpu-model
+  "Validates that GPU model entered is supported on the pool"
+  [pool-name {:keys [env]}]
+  (when (and (get env "COOK_GPU_MODEL")
+             (not (contains? (get-gpu-models-on-pool (config/valid-gpu-models) pool-name) (get env "COOK_GPU_MODEL"))))
+    (throw (ex-info (str "The following GPU model is not supported: " (get env "COOK_GPU_MODEL")) {}))))
```

Comment:
Seems worth it to extract a variable (in a `let`) for `(get env "COOK_GPU_MODEL")`, since we use it 3 times.